packages/block-library/src/button/edit.native.js

```diff
@@ -306,104 +313,107 @@ class ButtonEdit extends Component {
 				: placeholder || __( 'Add text…' );
 
 		return (
-			<View style={ { flex: 1 } } onLayout={ this.onLayout }>
-				<View
-					style={ [
-						styles.container,
-						isSelected && {
-							borderColor: this.getBackgroundColor(),
-							borderRadius: outlineBorderRadius,
-							borderWidth: styles.button.borderWidth,
-						},
-					] }
+			<View style={ { flex: 1 } } onLayout={ this.onContainerLayout }>
+				<ColorBackground
+					borderRadiusValue={ borderRadiusValue }
+					backgroundColor={ this.getBackgroundColor() }
+					isSelected={ isSelected }
 				>
-					<ColorBackground
-						borderRadiusValue={ borderRadiusValue }
-						backgroundColor={ this.getBackgroundColor() }
-					>
-						<RichText
-							setRef={ ( richText ) => {
-								this.richTextRef = richText;
-							} }
-							placeholder={ placeholderText }
-							value={ text }
-							onChange={ this.onChangeText }
-							style={ {
-								...richTextStyle.richText,
-								color: textColor.color || '#fff',
-							} }
-							textAlign="center"
-							placeholderTextColor={ 'lightgray' }
-							identifier="content"
-							tagName="p"
-							minWidth={ minWidth }
-							maxWidth={ maxWidth }
-							id={ clientId }
-							isSelected={ isButtonFocused }
-							withoutInteractiveFormatting
-							unstableOnFocus={ () =>
-								this.onToggleButtonFocus( true )
-							}
-							__unstableMobileNoFocusOnMount={ ! isSelected }
-							selectionColor={ textColor.color || '#fff' }
+					{ isSelected && (
+						<View
+							pointerEvents="none"
+							style={ [
+								styles.outline,
+								{
+									borderRadius: outlineBorderRadius,
+									borderWidth: styles.button.borderWidth,
+									borderColor: this.getBackgroundColor(),
+								},
+							] }
 						/>
-					</ColorBackground>
-
-					{ isButtonFocused && (
-						<BlockControls>
-							<ToolbarGroup>
-								<ToolbarButton
-									title={ __( 'Edit image' ) }
-									icon={ link }
-									onClick={ this.onToggleLinkSettings }
-								/>
-							</ToolbarGroup>
-						</BlockControls>
 					) }
-
-					<BottomSheet
-						isVisible={ isLinkSheetVisible }
-						onClose={ this.onToggleLinkSettings }
-						hideHeader
-					>
-						{ this.getLinkSettings( url, rel, linkTarget ) }
-						<BottomSheet.Cell
-							label={ __( 'Remove link' ) }
-							labelStyle={ styles.clearLinkButton }
-							separatorType={ 'none' }
-							onPress={ this.onClearSettings }
-						/>
-					</BottomSheet>
-
-					<InspectorControls>
-						<PanelBody title={ __( 'Border Settings' ) }>
-							<RangeControl
-								label={ __( 'Border Radius' ) }
-								minimumValue={ MIN_BORDER_RADIUS_VALUE }
-								maximumValue={ MAX_BORDER_RADIUS_VALUE }
-								value={ borderRadiusValue }
-								onChange={ this.onChangeBorderRadius }
-								separatorType="none"
+					<RichText
+						setRef={ ( richText ) => {
```

Comment:
Hmm, just thinking if it would be a bit more performant when that function would be the class method.